### DISCUSSION TITLE
wavf2 issue

### DISCUSSION THREAD

**Discussion by ramdaggumati** [aws/aws-cli#8834] [Q&A]
so when run the command in my aws cli:
`aws service-quotas get-service-quota --service-code wafv2 --quota-code L-2EC3DE7B`

it returns me the error:**NoSuchResourceException** but still i can find it on aws console 

note the aws account id and the region is same for cli and the console

**Comment by uldyssian-sh** [comment]
NoSuchResourceException from `service-quotas get-service-quota` means one of these is true:

- the **service code** you used (`wafv2`) is not the one Service Quotas expects in your Region, or
- the **quota code** (`L-2EC3DE7B`) is not a quota code for that service in that Region, or
- you are calling Service Quotas in the **wrong Region** (Service Quotas is Region-scoped), so the quota you see in the console is being looked up in a different Region than your CLI request.

The fix is to discover the correct service/quota codes for the exact Region you are using, then call get-service-quota with those exact values.

Step 1: confirm you are querying the same Region as the console

aws configure get region

Then explicitly set it in the commands below (replace us-east-1 with the Region where you see it in the console).

Step 2: find the correct service code in that Region (WAF may show up as `waf` or `wafv2` depending on integration)

aws service-quotas list-services \
  --region us-east-1 \
  --query "Services[?contains(ServiceName, 'WAF')].[ServiceName,ServiceCode]" \
  --output table

Step 3: list the quota codes for that service code in that Region and locate the one you see in the console

aws service-quotas list-service-quotas \
  --region us-east-1 \
  --service-code wafv2 \
  --query "Quotas[].{Name:QuotaName,Code:QuotaCode,Value:Value}" \
  --output table

(If `wafv2` returns nothing, rerun Step 3 with `--service-code waf` using the code you found in Step 2.)

Step 4: once you have the correct quota code for that Region, call get-service-quota

aws service-quotas get-service-quota \
  --region us-east-1 \
  --service-code <SERVICE_CODE_FROM_STEP_2> \
  --quota-code <QUOTA_CODE_FROM_STEP_3>

Why the console can “show it” while CLI fails:

The console often shows quotas for a selected service/Region, but your CLI call will fail if you ask for a quota code that is not registered under that service code in the Region you’re querying. The Service Quotas API is strict: service code + quota code + Region must match exactly.

**Reply by uldyssian-sh** [reply]
Thanks! If this helps, please consider marking it as “Helpful” or “Accepted” 😊
